pages/sorting.tsx

```diff
@@ -0,0 +1,251 @@
+import useLocalState from '../hooks/useLocalState';
+import useSupportsClipboardRead from '../hooks/useSupportsClipboardRead';
+
+export default function Sorting() {
+  const supportsClipboardRead = useSupportsClipboardRead();
+  const [inputText, setInputText] = useLocalState<string>({
+    key: 'inputTextLocalState',
+    defaultValue: '',
+  });
+
+  const [sortStyle, setSortStyle] = useLocalState<string>({
+    key: 'sortStyleLocalState',
+    defaultValue: 'ascending',
+  });
+
+  const [separator, setSeparator] = useLocalState<string>({
+    key: 'separatorLocalState',
+    defaultValue: ',',
+  });
+
+  const [outputText, setOutputText] = useLocalState<string>({
+    key: 'outputTextLocalState',
+    defaultValue: '',
+  });
+
+  const [toastOpen, setToastOpen] = useState<boolean>(false);
+  const [toastMessage, setToastMessage] = useState<string>('');
+  const [toastSeverity, setToastSeverity] =
+    useState<ToastProps['severity']>('success');
+
+  const handleSortStyleChange = (event: SelectChangeEvent) => {
+    setSortStyle(event.target.value);
+  };
+
+  const handleSeparatorChange = (event: SelectChangeEvent) => {
+    setSeparator(event.target.value);
+  };
+
+  const handleClear = () => {
+    setInputText('');
+    setOutputText('');
+  };
+
+  const handlePaste = async () => {
+    const textCopied = await navigator.clipboard.readText();
+    setInputText(textCopied);
+    setOutputText(decodeURIComponent(textCopied));
+  };
+
+  const handleSort = () => {
+    if (inputText === '') {
+      setToastMessage('Please enter some text to sort.');
+      setToastSeverity('error');
+      setToastOpen(true);
+      return;
+    }
+
+    if (separator === '') {
+      setToastMessage('Please enter a separator.');
+      setToastSeverity('error');
+      setToastOpen(true);
+      return;
+    }
+
+    let inputTextArray;
+
+    if (separator === 'lineBreak') {
+      inputTextArray = inputText.split(/\r\n|\n\r|\n|\r/);
+      setSeparator('\n');
+    } else {
+      inputTextArray = inputText.split(separator);
+    }
+    const outputTextArray =
+      sortStyle === 'reverse'
+        ? inputTextArray.reverse()
+        : inputTextArray.sort((a, b) => {
+            if (sortStyle === 'ascending') {
+              return a.localeCompare(b);
+            }
+            if (sortStyle === 'descending') {
+              return b.localeCompare(a);
+            }
+            if (sortStyle === 'ascending2') {
+              if (a.length === b.length) {
+                return a.localeCompare(b);
+              }
+              return a.length - b.length;
+            }
+            if (sortStyle === 'descending2') {
+              if (a.length === b.length) {
+                return b.localeCompare(a);
+              }
+              return b.length - a.length;
+            }
+            return 0;
+          });
+
+    setOutputText(outputTextArray.join(separator));
+    if (separator === '\n') setSeparator('lineBreak');
```

Comment:
Here is the cause of the issue requiring line break to be sorted twice. I think it'd be  best to store `separator` as an object with title and filter values, something like this:

```js
const separators = [
  { title: 'Line Break', filter: '\n' },
  // ...
];
```

That way we can keep the logic separate from the displayed text, and not have to do any juggling of that value to make things work.

Reply:
Let me try to fix this.

Reply:
Resolved the issue, seems like I just needed a little break from the code to fix it.